node/src/transition_frontier/genesis/transition_frontier_genesis_config.rs

```diff
@@ -143,67 +150,33 @@ impl GenesisConfig {
-                        .into_iter()
-                        .map(|(n, h)| (n, h.to_field()))
-                        .collect::<Vec<_>>(),
-                );
-                let (_mask, staking_epoch_total_currency) =
-                    Self::build_ledger_from_accounts_and_hashes(
-                        staking_epoch_data
-                            .accounts
-                            .into_iter()
-                            .map(|acc| (&acc).into()),
-                        staking_epoch_data
-                            .hashes
-                            .into_iter()
-                            .map(|(n, h)| (n, h.to_field()))
-                            .collect::<Vec<_>>(),
-                    );
-                let (_mask, next_epoch_total_currency) =
-                    Self::build_ledger_from_accounts_and_hashes(
-                        next_epoch_data
-                            .accounts
-                            .into_iter()
-                            .map(|acc| (&acc).into()),
-                        next_epoch_data
-                            .hashes
-                            .into_iter()
-                            .map(|(n, h)| (n, h.to_field()))
-                            .collect::<Vec<_>>(),
-                    );
-
-                let load_result = GenesisConfigLoaded {
-                    constants,
-                    genesis_ledger_hash: ledger_hash,
-                    genesis_total_currency,
-                    genesis_producer_stake_proof: genesis_producer_stake_proof(&mask),
-                    staking_epoch_ledger_hash: staking_epoch_data.ledger_hash.clone(),
-                    staking_epoch_total_currency,
-                    next_epoch_ledger_hash: next_epoch_data.ledger_hash.clone(),
-                    next_epoch_total_currency,
-                    staking_epoch_seed: staking_epoch_data.seed,
-                    next_epoch_seed: next_epoch_data.seed,
-                };
-                (mask, load_result)
+                let prebuilt = PrebuiltGenesisConfig::read(&mut bytes.as_ref())?;
+                prebuilt.load()
             }
             Self::DaemonJson(config) => {
                 let constants = config
                     .genesis
                     .as_ref()
                     .map_or(PROTOCOL_CONSTANTS, |genesis| genesis.protocol_constants());
                 let ledger = config.ledger.as_ref().ok_or(GenesisConfigError::NoLedger)?;
+                let config_name = ledger.ledger_name();
+                let cache_filename =
+                    openmina_cache_path(format!("prebuilt_configs/{}.bin", config_name)).unwrap();
+                let filename_str = cache_filename
+                    .clone()
+                    .into_os_string()
+                    .into_string()
+                    .unwrap();
+                if cache_filename.is_file() {
+                    let fd = std::fs::File::open(cache_filename.clone()).unwrap();
```

Comment:
Lets turn this unwrap call into a returned error

Reply:
Done.